sklearn/metrics/_plot/confusion_matrix.py

```diff
@@ -61,7 +61,7 @@ def plot(self, include_values=True, cmap='viridis',
 
         values_format : str, default=None
             Format specification for values in confusion matrix. If `None`,
-            the format specification is '.2g'.
+            the format specification is 'd' if < 1e7, '.2g' otherwise.
```

Comment:
That's not the condition, it should be (log10(x) > 7)

But I would just use 

>             the format specification is 'd' or '.2g', whichever is shorter

Reply:
Ahh I see it's not clear now. 

I was referring to the value of that specific number. If this value is larger than or equal to 1e7 it will be in '.2g' format, which is essentially this log10(x) check. But I understand it's not really clear like this, I'll update it.